### PR TITLE
[IMP] component/tags: add inline css tag

### PR DESCRIPTION
**Pull request by brboi**
This add an important feature: defining completely standalone owl components, with the template/style and javascript code together.

### CODE REVIEW

**Comment by brboi** [comment]
I just reintroduced existing code, but IMO we could do better because we never cleanup the `<style/>` elements inside the document's head.

File: src/component/style.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+export function applyStyles(ComponentClass: typeof Component) {
+  const toApply: { [componentName: string]: string } = {};
+  while (ComponentClass && ComponentClass.style) {
+    if (ComponentClass.hasOwnProperty("style")) {
+      toApply[ComponentClass.name] = ComponentClass.style;
+      delete (ComponentClass as any).style;
+    }
+    ComponentClass = Object.getPrototypeOf(ComponentClass);
+  }
+  for (const [componentName, styleId] of Object.entries(toApply).reverse()) {
```
Comment (brboi) [inline src/component/style.ts:27]: reversing is new here. Any thoughts on that @ged-odoo ?

File: src/component/style.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+function activateSheet(id: string, name: string) {
+  const sheet = globalStylesheets[id];
+  if (!sheet) {
+    throw new Error(
+      `Invalid css stylesheet for component '${name}'. Did you forget to use the 'css' tag helper?`
+    );
+  }
+  sheet.setAttribute("component", name);
```
Comment (ged-odoo) [inline src/component/style.ts:39]: we should probably use the `data-component` attribute instead

File: src/component/component_node.ts [hunk 3]
```diff
@@ -89,6 +90,7 @@ export class ComponentNode<T extends typeof Component = any> implements VNode<Co
     applyDefaultProps(props, C);
     this.component = new C(props, app.env, this) as any;
     this.renderFn = app.getTemplate(C.template).bind(null, this.component, this);
+    applyStyles(C);
```
Comment (ged-odoo) [inline src/component/component_node.ts:93]: i would still wrap this in a if: `if (C.styles) {...}`

File: src/component/style.ts (hunk 1 above)
Comment (ged-odoo) [reply src/component/style.ts:27]: toapply => list, push in it, then pop in while loop

**Review by ged-odoo — APPROVED** [review]

**Pull request merged**